### DISCUSSION TITLE
Improving experience using keyboard for table creation.

### DISCUSSION THREAD

**Discussion by MJDeligan** [supabase/supabase#8496] [General]
Hey there, I wasn't sure if I should make an issue for this, but due to the subjectivity it's probably more of a discussion. Anyways...

I had to create a couple of tables using a Laptop without a mouse and noticed that the table editor does not work well when using only a keyboard. I found this discussion on [keyboard shortcuts](https://github.com/supabase/supabase/discussions/234) but couldn't find any related issues or commits and a lot the shortcuts do not seem to work for me.

The key annoyances I faced were:
- Pressing the 'Add Column'-Button (by pressing Enter while it is the active item) keeps the focus on the button, meaning the user has to shift-tab through all the input elements of the column and then back.
- The removal button cannot be selected by tabbing, meaning it is (to my knowledge) not possible to remove a column using only the keyboard.
- Removing a column while the active element is an element in the column causes the focus to be on the editor dialog, which forces users to then tab through all fields to get back down.
- The column reordering functionality seems to be inaccessible using a keyboard.

___
I have done some local development and improved on the first 3 points.

Here are the relevant commits: [https://github.com/supabase/supabase/compare/master...MJDeligan:supabase:master](https://github.com/supabase/supabase/compare/master...MJDeligan:supabase:master). _This was the first time I've worked with React so feel free to roast my code._
___

### Adding column

**Before:** Focus stays on button.
![col-add-before](https://user-images.githubusercontent.com/48515433/185484679-7ab7554c-688e-466b-b168-5444974f26af.gif)
<br>
**After:** Focus on the column name input of the created column input elements.
![col-add](https://user-images.githubusercontent.com/48515433/185485101-ed07596e-d403-4330-a42c-16cd03250503.gif)

### Removing using keyboard
**Before:** You couldn't select the remove "button" via keyboard.
![remove-before-skip](https://user-images.githubusercontent.com/48515433/185485375-5c969815-df72-40f8-b74e-cede0a09d80e.gif)
<br>
And if you did you'd have to tab all the way down.
![remove-before-focus](https://user-images.githubusercontent.com/48515433/185485528-df62b944-e3f1-47c1-bc16-a92866e8b462.gif)
<br>
**After:** You can select the remove "button" and when the column is removed while the active element is inside it the focus will be set to the "Add column" button
![remove](https://user-images.githubusercontent.com/48515433/185486061-8d340699-560c-4459-acf7-5b56d9d2684c.gif)
___
This is of course only how I would expect keyboard interaction to behave and I'd like to know other people's preferences and suggestions to further facilitate keyboard-only-usage.

Maybe there's also better ways to do these things in the current version. If so please, please, please let me know.

**Comment by saltcod** [comment]
Yep you're right @MJDeligan.   Adding this to a (giant) internal table improvement task. Thanks for the feedback!
